src/Git.jl

```diff
@@ -63,9 +63,9 @@ function iscommit(sha1s::Vector; dir="")
 end
 
 immutable State
-    head::String
-    index::String
-    work::String
+    head::AbstractString
```

Comment:
Oh good to know, thanks. I didn't want to introduce a Compat dependency just for that one occurrence, so maybe I'll do something like
```julia
if !isdefined(Core, :String)
    const String = ASCIIString
end
```
That should work for 0.4. I think DataStreams does that.